src/server/main_service.cc

```diff
@@ -1280,7 +1280,8 @@ bool StartMultiEval(DbIndex dbid, CmdArgList keys, ScriptMgr::ScriptParams param
   Transaction::MultiMode multi_mode = DetermineMultiMode(params);
 
   // Check if eval is already part of a running multi transaction
-  if (trans->GetMultiMode() != Transaction::NOT_DETERMINED) {
+  if ((trans->GetMultiMode() != Transaction::NOT_DETERMINED) &&
+      trans->GetMultiMode() != Transaction::NON_ATOMIC) {
```

Comment:
can you explain what this condition checks?

Reply:
StartMultiEval is called from EvalInternal, so this function might be called in the scope of a multi transaction or just a single eval
When running inside multi this function should not do anything as the flow is done in the exec flow - StartMultiExec.
When eval is not inside multi we need call the trans->StartMulti* function based on the eval mode.
In this if we check if we are inside a multi transaction, which actually need to be != Transaction::NOT_DETERMINED.
I can actually discard my change here of Transaction::NON_ATOMIC as it from the last commit, as we should not get to the state where running eval in non atomic multi mode, I return will error in such case.